benchmarks/hallucination/mmlu/run.py

```diff
@@ -0,0 +1,114 @@
+import pickle
+from string import ascii_uppercase as auc
+
+from datasets import (
+    get_dataset_config_names,
+    load_dataset,
+)
+import numpy as np
+from transformers import (
+    GPT2LMHeadModel,
+    GPT2TokenizerFast,
+)
+
+from fortuna.hallucination import HallucinationMulticalibrator
+from fortuna.hallucination.utils import string_cleaner
+from fortuna.metric.classification import accuracy
+
+SEED = 0
+CALIB_FRAC = 0.8
+
+if __name__ == "__main__":
+    device = "cuda"
+    model_id = "gpt2-large"
+    model = GPT2LMHeadModel.from_pretrained(model_id).to(device)
+    tokenizer = GPT2TokenizerFast.from_pretrained(model_id)
+
+    # download and prepare data
+    task_list = get_dataset_config_names("lukaemon/mmlu")
+    dataset_list = [
+        (
+            load_dataset(
+                "lukaemon/mmlu",
+                task,
+            ),
+            task,
+        )
+        for task in task_list
+    ]
+
+    answer_map = {a: i for i, a in enumerate(auc)}
+    samples = []
+    for datasets, task in dataset_list:
+        for dataset_key, dataset in datasets.items():
+            for sample in dataset:
+                samples.append(
+                    dict(
+                        question=string_cleaner(sample["input"]),
+                        choices=[sample[letter] for letter in ["A", "B", "C", "D"]],
+                        targets=answer_map[sample["target"]],
+                    )
+                )
+
+    # shuffle and split
+    rng = np.random.default_rng(seed=SEED)
+    tot_size = len(samples)
+    perm = rng.choice(tot_size, tot_size, replace=False)
+    samples = [samples[i] for i in perm]
+
+    calib_size = int(np.ceil(CALIB_FRAC * tot_size))
+    calib_choices, calib_questions, calib_targets = [], [], []
+    test_choices, test_questions, test_targets = [], [], []
+    for i, sample in enumerate(samples):
+        if i < calib_size:
+            calib_questions.append(sample["question"])
+            calib_choices.append(sample["choices"])
+            calib_targets.append(sample["targets"])
+        else:
+            test_questions.append(sample["question"])
+            test_choices.append(sample["choices"])
+            test_targets.append(sample["targets"])
+
+    # calibrate
+    calibrator = HallucinationMulticalibrator(
+        generative_model=model, tokenizer=tokenizer
+    )
+
+    status = calibrator.fit(
+        texts=calib_choices,
+        contexts=calib_questions,
+        targets=calib_targets,
+    )
+
+    with open("fitted_calibrator.pth", "wb") as filehandler:
+        pickle.dump(calibrator, filehandler, -1)
```

Comment:
is this important? do we need to load that later?

Reply:
Yes. We might want to use the calibrated model in a demo where the user brings in their own answers.